pkg/util/test/util.go

```diff
@@ -2578,6 +2610,255 @@ func ValidateAlertManagerDisabled(pxImageList map[string]string, cluster *corev1
+		return fmt.Errorf("envoy image mismatch, image: %s, expected: %s",
+			metricsCollectorDep.Spec.Template.Spec.Containers[1].Image,
+			envoyImageName)
+	}
+
+	// Wait for the daemonset to become online
+	logrus.Info("Validate px-telemetry-phonehome daemonset and images")
+	if err := appops.Instance().ValidateDaemonSet("px-telemetry-phonehome", cluster.Namespace, timeout); err != nil {
+		return err
+	}
+	telemetryPhonehomeDs, err := appops.Instance().GetDaemonSet("px-telemetry-phonehome", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Verify collector container image inside px-telemetry-phonehome
+	logUploaderImageName, ok := pxImageList["logUploader"]
+	if !ok {
+		return fmt.Errorf("failed to find image for log uploader")
+	}
+	logUploaderImageName = util.GetImageURN(cluster, logUploaderImageName)
+
+	if telemetryPhonehomeDs.Spec.Template.Spec.Containers[0].Image != logUploaderImageName {
+		return fmt.Errorf("log uploader image mismatch, image: %s, expected: %s",
+			telemetryPhonehomeDs.Spec.Template.Spec.Containers[0].Image,
+			logUploaderImageName)
+	}
+
+	// Verify envoy container image inside px-telemetry-phonehome
+	if telemetryPhonehomeDs.Spec.Template.Spec.Containers[1].Image != envoyImageName {
+		return fmt.Errorf("envoy image mismatch, image: %s, expected: %s",
+			telemetryPhonehomeDs.Spec.Template.Spec.Containers[1].Image,
+			envoyImageName)
+	}
+
+	// Verify telemetry roles
+	_, err = rbacops.Instance().GetRole("px-telemetry", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Verify telemetry secrets
+	_, err = coreops.Instance().GetSecret("pure-telemetry-certs", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Verify telemetry rolebindings
+	_, err = rbacops.Instance().GetRoleBinding("px-telemetry", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Verify telemetry configmaps
+	_, err = coreops.Instance().GetConfigMap("px-telemetry-collector", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+	_, err = coreops.Instance().GetConfigMap("px-telemetry-collector-proxy", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+	_, err = coreops.Instance().GetConfigMap("px-telemetry-phonehome", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+	_, err = coreops.Instance().GetConfigMap("px-telemetry-phonehome-proxy", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+	_, err = coreops.Instance().GetConfigMap("px-telemetry-register", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	_, err = coreops.Instance().GetConfigMap("px-telemetry-register-proxy", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	_, err = coreops.Instance().GetConfigMap("px-telemetry-tls-certificate", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Verify telemetry serviceaccounts
+	_, err = coreops.Instance().GetServiceAccount("px-telemetry", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
```

Comment:
are we able to validate the clusterrole and clusterrolebinding as well?

Reply:
Done